src/compiler/transformers/declarations.ts

```diff
@@ -1595,7 +1595,7 @@ export function transformDeclarations(context: TransformationContext) {
                     needsScopeFixMarker = oldNeedsScopeFix;
                     resultHasScopeMarker = oldHasScopeFix;
                     const mods = ensureModifiers(input);
-                    return cleanup(factory.updateModuleDeclaration(
+                    return cleanup(factory.updateModuleDeclarationAndKeyword(

```

Comment:
We don't really need another factory method for this. The same thing can be accomplished fairly trivially inline in this file, since its only needed here:

```ts
function updateModuleDeclarationAndKeyword(
    node: ModuleDeclaration,
    modifiers: readonly ModifierLike[] | undefined,
    name: ModuleName,
    body: ModuleBody | undefined
) {
    const updated = factory.updateModuleDeclaration(node, modifiers, name, body);
    // no further action for module augmentations or if the node already uses `namespace`
    if (isAmbientModule(updated) || updated.flags & NodeFlags.Namespace) {
        return updated;
    }

    const fixed = factory.createModuleDeclaration(
        output.modifiers,
        output.name,
        output.body,
        output.flags | NodeFlags.Namespace
    );
    setOriginalNode(fixed, updated);
    return fixed;
}
```

Reply:
Good point - I have updated the code to use an inline function instead.